### PR TITLE
New SDK update for Automation.

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
## Description
Contains changes introduced in the following spec:
Azure/azure-rest-api-specs#3624

Other changes:
Increased the version of the SDK - 3.4.0-preview.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by Francisco-Gamino** [comment]
Adding @vrdmr for FYI.

File: src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs [hunk 1]
```diff
@@ -12,11 +12,11 @@ namespace Microsoft.Azure.Management.Automation.Models
 {
 
     /// <summary>
-    /// Defines values for StartType.
+    /// Defines values for SyncType.
     /// </summary>
-    public static class StartType
```
Comment (dsgouda) [inline src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs]: This is a breaking change for SDK users, please update the version number to `3.4.0-preview`

File: src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs [hunk 2]
```diff
@@ -467,7 +467,7 @@ public void CanCreateSourceControlSyncJob()
                     Assert.Equal(retrievedSyncJob.ProvisioningState, syncJob.ProvisioningState);
                     Assert.Equal(retrievedSyncJob.StartTime, syncJob.StartTime);
                     Assert.Equal(retrievedSyncJob.EndTime, syncJob.EndTime);
-                    Assert.Equal(retrievedSyncJob.StartType, syncJob.StartType);
+                    //Assert.Equal(retrievedSyncJob.SyncType, syncJob.SyncType);
```
Comment (dsgouda) [inline src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs]: Why is this commented

File: src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs (hunk 1 above)
Comment (dsgouda) [reply src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs]: Please update the aseembly file version [here ](https://github.com/Azure/azure-sdk-for-net/blob/9d719904e51dfb1efc1893575d7d3352964a8b99/src/SDKs/Automation/Management.Automation/Properties/AssemblyInfo.cs#L11)

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs (hunk 2 above)
Comment (Francisco-Gamino) [reply src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs]: The web service changes have not reached the region that is used to record the tests. If you prefer, I can skip the test and rerecord it for the next SDK update. Let me know. Thanks.

File: src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs (hunk 1 above)
Comment (Francisco-Gamino) [reply src/SDKs/Automation/Management.Automation/Generated/Models/SyncType.cs]: Fixed.

File: src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs (hunk 2 above)
Comment (Francisco-Gamino) [reply src/SDKs/Automation/Automation.Tests/ScenarioTests/AutomationTests.cs]: Fixed.

**Comment by Francisco-Gamino** [comment]
@dsgouda : Could you please take another look? Thanks!

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by Francisco-Gamino** [comment]
Awesome, thanks @dsgouda !

**Pull request merged**